OCD-4081: Updates to Questionable URL Report

kekey1 opened the pull request.

kekey1 committed:
fix: Set empty response text when there is none

[#OCD-4081]

kekey1 committed:
feat: Add stack trace for unknown url failures

The plan is to add the stack trace for more information (hopefully)
about why the fhirtest url fails in production only.

[#OCD-4081]

kekey1 committed:
refactor: Use system user ID for url check data; use newer StringUtils

I think the last_modified_user value in the url_check_result table was
using the ID of the most recently logged-in user. It should probably
just be our system user ID (-1).

[#OCD-4081]

kekey1 committed:
feat: Use HttpClient instead of AsyncHttpClient for URL Calls

The thought was that the AsyncHttpClient library hasn't been updated in
over a year since sometime in 2021. I wanted to try changing the library
used to be one that is at least regularly updated. We cannot use the
latest HttpClient v5 without also upgrading our version of Spring so we
are just trying HttpClient v4 here.

[#OCD-4081]

kekey1 committed:
Merge remote-tracking branch 'upstream/staging' into OCD-4081

kekey1 committed:
Merge branch 'staging' into OCD-4081

kekey1 committed:
build: Remove dependency on asynchttpclient

[#OCD-4081]

kekey1 committed:
Merge branch 'staging' into OCD-4081

kekey1 committed:
feat: Remove checking of URLS related to inactive certificates

[#OCD-4081]

kekey1 committed:
fix: Only include active certificates in admin/onc url report

[#OCD-4081]

kekey1 committed:
test: Add test that can be used for specific URL research

[#OCD-4081]

andlar commented on chpl/chpl-service/src/main/java/gov/healthit/chpl/scheduler/job/urlStatus/QuestionableUrlReportGenerator.java:
```diff
@@ -170,8 +170,11 @@ private List<FailedUrlResult> filterUrls(List<FailedUrlResult> badUrls, JobExecu
                 .collect(Collectors.toList());
+        } else {
+            return badUrls.stream()
+                .filter(badUrl -> isNotListingUrl(badUrl) || (isUrlRelatedTo2015Edition(badUrl) && isUrlRelatedToActiveListing(badUrl)))
```
Given the change in the DAO related to the query against Active Certificates, does the part after the `||` always evaluate to `true` now, for any URL that is a ListingUrl? So the `isNotListing` call will be true for non-Listing URLs, and the second clause will be true for all Listing URLs, so this whole filter will always be `true || false` or `false || true`, which means the filter never filters anything?

kekey1 replied:
I thought so, too, and only added this in after I found it was not the case. When we collect status data for URLs related to listings, we are using only URLs for listings that are Active/Suspended. Later when we get into creating this report we take each failed URL and figure out what listing(s) it relates to so we can include each one in the report. Sometimes the failed URL that was used for an Active/Suspended listing is ALSO used in an Inactive listing so this filter is meant to get rid of those.

kekey1 committed:
Merge branch 'staging' into OCD-4081

kekey1 committed:
Merge branch 'staging' into OCD-4081

kekey1 committed:
Merge branch 'staging' into OCD-4081

kekey1 committed:
Merge branch 'staging' into OCD-4081

kekey1 committed:
Merge branch 'staging' into OCD-4081

kekey1 committed:
fix: Remove ATL bullet from ACB-specific report

ACBs are not able to change ATL URL data and the ATLs are not included
in the CSV attachment for the ACB-specific report so in order to be
consistent we should not include ATLs in the email body either.

[#OCD-4081]

kekey1 committed:
Merge branch 'staging' into OCD-4081
